### PR TITLE
Separate personal and group chat

### PR DESCRIPTION
**Pull request by Vladyslav-Kuksiuk**
This PR separates personal and group chat creation.

Personal and group chat creation flow models:
image

### CODE REVIEW

File: model/src/main/proto/spine_examples/chatspn/chat/chat.proto [hunk 1]
```diff
@@ -47,6 +47,14 @@ message Chat {
     // List of the chat members.
     repeated spine.core.UserId member = 2 [(required) = true, (distinct) = true];
 
+    // The type of the chat.
+    ChatType type = 3;
+
+    enum ChatType {
+        PERSONAL = 0;
+        GROUP = 1;
+    }
```
Comment (Artem-Semenov-dev) [inline model/src/main/proto/spine_examples/chatspn/chat/chat.proto:56]: I think a more specific and broader explanation is needed here.

File: server/src/test/java/io/spine/examples/chatspn/server/chat/ChatMembersProjectionTest.java [hunk 2]
```diff
@@ -26,17 +26,18 @@
...
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
-import static io.spine.testing.TestValues.randomString;
+import static io.spine.examples.chatspn.server.given.ChatMembersProjectionTestEnv.chatMembersFrom;
+import static io.spine.examples.chatspn.server.given.ChatTestEnv.createGroupChatCommand;
+import static io.spine.examples.chatspn.server.given.ChatTestEnv.createPersonalChatCommand;
 
 @DisplayName("`ChatMembersProjection` should")
 class ChatMembersProjectionTest extends ContextAwareTest {
```
Comment (Artem-Semenov-dev) [inline server/src/test/java/io/spine/examples/chatspn/server/chat/ChatMembersProjectionTest.java:43]: Looks final to me.

**Review by Artem-Semenov-dev — COMMENTED** [review]
@Vladyslav-Kuksiuk please see my comments.

**Review by armiol — COMMENTED** [review]
I still don't understand why chat-for-two is different from chat-for-many. Why isn't chat-for-two just a sub-type of chat-for-many?

Please explain that in the documentation of the corresponding Proto types (it's probable where Artem asked you to document things).

File: server/src/test/java/io/spine/examples/chatspn/server/given/ChatTestEnv.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.chatspn.server.given;
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/chatspn/server/given/ChatTestEnv.java:27]: If the correponding test class (in this case, it's `ChatTest`) resides in the package like `foo.bar`, its test environment should go to `foo.bar.given` (with the appropriate `package-info.java`, as always).

Please review.

**Review by armiol — CHANGES_REQUESTED** [review]
@Vladyslav-Kuksiuk LGTM for the most part. Please see my comments.

File: server/src/test/java/io/spine/examples/chatspn/server/given/ChatTestEnv.java (hunk 3 above)
Comment (Vladyslav-Kuksiuk) [reply server/src/test/java/io/spine/examples/chatspn/server/given/ChatTestEnv.java:27]: I have the same problem in every TestEnv, I'll fix them in [improve-tests-readability branch](https://github.com/spine-examples/ChatSPN/tree/improve-tests-readability).

**Review by armiol — APPROVED** [review]
@Vladyslav-Kuksiuk LGTM with a single comment to address.

**Pull request merged**